### DISCUSSION TITLE
LDAP or SSO authentication

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#506] [Q&A] [answered]
Is it possible to integrate LDAP( AD) authentication with either UI or rest endpoint(server side). Also any possible way of integrating SSO authentication.

**Comment by drakkan** [comment, marked as answer]
Hi,

you can use external authentication or an auth plugin (auth plugins are supported in the main development branch).  Please take a look at the following examples:

https://github.com/drakkan/sftpgo/tree/main/examples/ldapauthserver
https://github.com/drakkan/sftpgo/tree/main/examples/ldapauth
https://www.youtube.com/watch?v=M5UcJI8t4AI

You can use them as starting point.

My long term plan is to provide an auth plugin that allows to handle LDAP (AD), system users ecc.. We could add a user type enum to the REST API/web admin which the authentication plugin uses as a hint to understand if it needs to lookup for LDAP users, system users etc.

However I'm not a Windows guy, I have limited Windows experience and only a Windows VM, I have no concrete plan for this feature. A PR or a sponsorship could help. Thank you

**Reply by rashmimaharana** [reply]
thanks @drakkan for the info.
I have one concerns, i will create a separate discussion for that.

**Reply by drakkan** [reply]
Ok, let me know, thank you
